common/xfer/pipes.go

```diff
@@ -93,7 +93,7 @@ func (p *pipe) CopyToWebsocket(end io.ReadWriter, conn Websocket) error {
 	p.mtx.Unlock()
 	defer p.wg.Done()
 
-	errors := make(chan error, 1)
+	errors := make(chan error, 2)
```

Comment:
The goroutines both post their errors to the channel, which if you close() the pipe may not get read from.  Therefore it needs up to 2 slots free.

Its ugly.  I need to come back to this at some point in the future and make this all a bit more sane.